external-crates/move/crates/move-package/tests/test_lock_file.rs

```diff
@@ -241,29 +241,21 @@ fn test_update_managed_address() {
 
     // Read lock file and check contents.
     let mut lock_file = File::open(lock_path).unwrap();
-    let contents = {
-        let mut buf = String::new();
-        let _ = lock_file.read_to_string(&mut buf);
-        buf
-    };
+    let envs = ManagedPackage::read(&mut lock_file)
+        .unwrap()
+        .iter()
+        .map(|(key, value)| format!("{}={:#?}", key, value))
+        .collect::<Vec<String>>()
+        .join("\n");
 
     let expected = expect![[r#"
-        # @generated by Move, please check-in and do not edit manually.
-
-        [move]
-        version = 1
-        manifest_digest = "42"
-        deps_digest = "7"
-
-        [env]
-
-        [env.default]
-        chain-id = "35834a8a"
-        original-published-id = "0x123"
-        latest-published-id = "0x456"
-        published-version = "2"
-    "#]];
-    expected.assert_eq(&contents);
+        default=ManagedPackage {
+            chain_id: "35834a8a",
+            original_published_id: "0x123",
+            latest_published_id: "0x456",
+            version: "2",
+        }"#]];
+    expected.assert_eq(&envs);
```

Comment:
nit: You could probably use `assert_debug_eq` here and not create the strings yourself.

```rust
let envs: Vec<_> = ManagedPackage::read(&mut lock_file)
    .unwrap()
    .into_iter()
    .collect();

let expected = expect![[r#"..."#]];
expected.assert_debug_eq(&envs);
```

Reply:
nice, missed converting this to `vec` trick